fe/fe-core/src/main/java/org/apache/doris/analysis/BinaryPredicate.java

```diff
@@ -307,6 +307,16 @@ private Type getCmpType() {
             return Type.LARGEINT;
         }
 
+        // Try parse varchar to bigint
+        if (t1 == PrimitiveType.BIGINT && t2 == PrimitiveType.VARCHAR) {
```

Comment:
Because we have `NormalizeBinaryPredicatesRule`. You could only consider slot in left and string Literal in right

Reply:
> Because we have `NormalizeBinaryPredicatesRule`. You could only consider slot in left and string Literal in right
但是我试了一下，如果我注释掉代码后，如果SQL这么写，还是转成Double了，虽然很少有人这么写SQL。
explain select count()
from app_rt_flow_poi_intent_uv_price_v2
where '20200730'=datekey
回头我看一下NormalizeBinaryPredicatesRule的原理。

Reply:
Yes. you are right. Don't need change this.